XMLFormatterModel/Update/Strategies/GitHubVersionRecieverStrategy.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Octokit;
+using PluginFramework.EventMessages;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using XmlFormatterModel.Update.Adapter;
+
+namespace XmlFormatterModel.Update.Strategies
+{
+    /// <summary>
+    /// Get the newest version from GitHub
+    /// </summary>
+    public class GitHubVersionRecieverStrategy : IVersionRecieverStrategy
+    {
+        /// <inheritdoc/>
+        public event EventHandler<BaseEventArgs> Error;
+
+        private readonly Regex regex;
+
+        /// <summary>
+        /// Create a new instance of this class
+        /// </summary>
+        public GitHubVersionRecieverStrategy()
+        {
+            regex = new Regex(@"([0-9]{1,}.[0-9]{1,}.[0-9]{1,})");
+        }
+
+        /// <inheritdoc/>
+        public async Task<List<IRelease>> GetReleases()
+        {
+            List<IRelease> returnReleases = new List<IRelease>();
+            GitHubClient client = new GitHubClient(new ProductHeaderValue("XanatosX"));
+            IReadOnlyList<Release> releases = await client.Repository.Release.GetAll("XanatosX", "XmlFormatter");
+            foreach (Release gitHubRelease in releases)
+            {
+                returnReleases.Add(new GitHubReleaseAdapter(gitHubRelease));
+            }
+            return returnReleases;
+        }
+
+        /// <inheritdoc/>
+        public async Task<IRelease> GetLatestRelease()
+        {
+            List<IRelease> releases = await GetReleases();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=5634024)